{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "id": "C4ze5dKVct59",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting mxnet\n",
      "  Using cached mxnet-1.7.0.post1-py2.py3-none-win_amd64.whl (33.0 MB)\n",
      "Requirement already satisfied: graphviz<0.9.0,>=0.8.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from mxnet) (0.8.4)\n",
      "Collecting numpy<1.17.0,>=1.8.2\n",
      "  Using cached numpy-1.16.6.zip (5.1 MB)\n",
      "Collecting requests<2.19.0,>=2.18.4\n",
      "  Using cached requests-2.18.4-py2.py3-none-any.whl (88 kB)\n",
      "Requirement already satisfied: idna<2.7,>=2.5 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from requests<2.19.0,>=2.18.4->mxnet) (2.6)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from requests<2.19.0,>=2.18.4->mxnet) (2020.12.5)\n",
      "Requirement already satisfied: chardet<3.1.0,>=3.0.2 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from requests<2.19.0,>=2.18.4->mxnet) (3.0.4)\n",
      "Collecting urllib3<1.23,>=1.21.1\n",
      "  Using cached urllib3-1.22-py2.py3-none-any.whl (132 kB)\n",
      "Building wheels for collected packages: numpy\n",
      "  Building wheel for numpy (setup.py): started\n",
      "  Building wheel for numpy (setup.py): finished with status 'error'\n",
      "  Running setup.py clean for numpy\n",
      "Failed to build numpy\n",
      "Installing collected packages: urllib3, requests, numpy, mxnet\n",
      "  Attempting uninstall: urllib3\n",
      "    Found existing installation: urllib3 1.26.3\n",
      "    Uninstalling urllib3-1.26.3:\n",
      "      Successfully uninstalled urllib3-1.26.3\n",
      "  Attempting uninstall: requests\n",
      "    Found existing installation: requests 2.25.1\n",
      "    Uninstalling requests-2.25.1:\n",
      "      Successfully uninstalled requests-2.25.1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\caell\\new_anaconda3\\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\caell\\\\AppData\\\\Local\\\\Temp\\\\pip-install-rg58yztv\\\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\caell\\\\AppData\\\\Local\\\\Temp\\\\pip-install-rg58yztv\\\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d 'C:\\Users\\caell\\AppData\\Local\\Temp\\pip-wheel-oh27e96x'\n",
      "       cwd: C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\n",
      "  Complete output (376 lines):\n",
      "  Running from numpy source directory.\n",
      "  C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\misc_util.py:476: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
      "    return is_string(s) and ('*' in s or '?' is s)\n",
      "  blas_opt_info:\n",
      "  blas_mkl_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries mkl_rt not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  blis_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries blis not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  openblas_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries openblas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "  get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "  customize GnuFCompiler\n",
      "  Could not locate executable g77\n",
      "  Could not locate executable f77\n",
      "  customize IntelVisualFCompiler\n",
      "  Could not locate executable ifort\n",
      "  Could not locate executable ifl\n",
      "  customize AbsoftFCompiler\n",
      "  Could not locate executable f90\n",
      "  customize CompaqVisualFCompiler\n",
      "  Could not locate executable DF\n",
      "  customize IntelItaniumVisualFCompiler\n",
      "  Could not locate executable efl\n",
      "  customize Gnu95FCompiler\n",
      "  Found executable C:\\Users\\caell\\new_Anaconda3\\Library\\mingw-w64\\bin\\gfortran.exe\n",
      "  Using built-in specs.\n",
      "  COLLECT_GCC=C:\\Users\\caell\\new_Anaconda3\\Library\\mingw-w64\\bin\\gfortran.exe\n",
      "  COLLECT_LTO_WRAPPER=C:/Users/caell/new_Anaconda3/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/lto-wrapper.exe\n",
      "  Target: x86_64-w64-mingw32\n",
      "  Configured with: ../gcc-5.3.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --with-gxx-include-dir=/mingw64/include/c++/5.3.0 --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,objc,obj-c++,fortran,ada --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --enable-version-specific-runtime-libs --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev5, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld\n",
      "  Thread model: posix\n",
      "  gcc version 5.3.0 (Rev5, Built by MSYS2 project)\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  atlas_3_10_blas_threads_info:\n",
      "  Setting PTATLAS=ATLAS\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries tatlas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  atlas_3_10_blas_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries satlas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  atlas_blas_threads_info:\n",
      "  Setting PTATLAS=ATLAS\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries ptf77blas,ptcblas,atlas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  atlas_blas_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries f77blas,cblas,atlas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  accelerate_info:\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "      Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "      Directories to search for the libraries can be specified in the\n",
      "      numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "      the ATLAS environment variable.\n",
      "    self.calc_info()\n",
      "  blas_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries blas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "      Blas (http://www.netlib.org/blas/) libraries not found.\n",
      "      Directories to search for the libraries can be specified in the\n",
      "      numpy/distutils/site.cfg file (section [blas]) or by setting\n",
      "      the BLAS environment variable.\n",
      "    self.calc_info()\n",
      "  blas_src_info:\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "      Blas (http://www.netlib.org/blas/) sources not found.\n",
      "      Directories to search for the sources can be specified in the\n",
      "      numpy/distutils/site.cfg file (section [blas_src]) or by setting\n",
      "      the BLAS_SRC environment variable.\n",
      "    self.calc_info()\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  'svnversion' is not recognized as an internal or external command,\n",
      "  operable program or batch file.\n",
      "  non-existing path in 'numpy\\\\distutils': 'site.cfg'\n",
      "  lapack_opt_info:\n",
      "  lapack_mkl_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries mkl_rt not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  openblas_lapack_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries openblas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "  get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "  customize GnuFCompiler\n",
      "  customize IntelVisualFCompiler\n",
      "  customize AbsoftFCompiler\n",
      "  customize CompaqVisualFCompiler\n",
      "  customize IntelItaniumVisualFCompiler\n",
      "  customize Gnu95FCompiler\n",
      "  Using built-in specs.\n",
      "  COLLECT_GCC=C:\\Users\\caell\\new_Anaconda3\\Library\\mingw-w64\\bin\\gfortran.exe\n",
      "  COLLECT_LTO_WRAPPER=C:/Users/caell/new_Anaconda3/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/lto-wrapper.exe\n",
      "  Target: x86_64-w64-mingw32\n",
      "  Configured with: ../gcc-5.3.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --with-gxx-include-dir=/mingw64/include/c++/5.3.0 --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,objc,obj-c++,fortran,ada --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --enable-version-specific-runtime-libs --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev5, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld\n",
      "  Thread model: posix\n",
      "  gcc version 5.3.0 (Rev5, Built by MSYS2 project)\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  openblas_clapack_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries openblas,lapack not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "  get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "  customize GnuFCompiler\n",
      "  customize IntelVisualFCompiler\n",
      "  customize AbsoftFCompiler\n",
      "  customize CompaqVisualFCompiler\n",
      "  customize IntelItaniumVisualFCompiler\n",
      "  customize Gnu95FCompiler\n",
      "  Using built-in specs.\n",
      "  COLLECT_GCC=C:\\Users\\caell\\new_Anaconda3\\Library\\mingw-w64\\bin\\gfortran.exe\n",
      "  COLLECT_LTO_WRAPPER=C:/Users/caell/new_Anaconda3/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/lto-wrapper.exe\n",
      "  Target: x86_64-w64-mingw32\n",
      "  Configured with: ../gcc-5.3.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --with-gxx-include-dir=/mingw64/include/c++/5.3.0 --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,objc,obj-c++,fortran,ada --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --enable-version-specific-runtime-libs --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev5, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld\n",
      "  Thread model: posix\n",
      "  gcc version 5.3.0 (Rev5, Built by MSYS2 project)\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  atlas_3_10_threads_info:\n",
      "  Setting PTATLAS=ATLAS\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries tatlas,tatlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in C:\\\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries tatlas,tatlas not found in C:\\\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries tatlas,tatlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries tatlas,tatlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "  <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  atlas_3_10_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries satlas,satlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in C:\\\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries satlas,satlas not found in C:\\\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries satlas,satlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries satlas,satlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "  <class 'numpy.distutils.system_info.atlas_3_10_info'>\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  atlas_threads_info:\n",
      "  Setting PTATLAS=ATLAS\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries ptf77blas,ptcblas,atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in C:\\\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries ptf77blas,ptcblas,atlas not found in C:\\\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries ptf77blas,ptcblas,atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries ptf77blas,ptcblas,atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "  <class 'numpy.distutils.system_info.atlas_threads_info'>\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  atlas_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries f77blas,cblas,atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in C:\\\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries f77blas,cblas,atlas not found in C:\\\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries f77blas,cblas,atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack_atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries f77blas,cblas,atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "  <class 'numpy.distutils.system_info.atlas_info'>\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  lapack_info:\n",
      "  No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "  customize MSVCCompiler\n",
      "    libraries lapack not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "      Lapack (http://www.netlib.org/lapack/) libraries not found.\n",
      "      Directories to search for the libraries can be specified in the\n",
      "      numpy/distutils/site.cfg file (section [lapack]) or by setting\n",
      "      the LAPACK environment variable.\n",
      "    self.calc_info()\n",
      "  lapack_src_info:\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "      Lapack (http://www.netlib.org/lapack/) sources not found.\n",
      "      Directories to search for the sources can be specified in the\n",
      "      numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n",
      "      the LAPACK_SRC environment variable.\n",
      "    self.calc_info()\n",
      "    NOT AVAILABLE\n",
      "  \n",
      "  c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py:274: UserWarning: Unknown distribution option: 'define_macros'\n",
      "    warnings.warn(msg)\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running config_cc\n",
      "  unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n",
      "  running config_fc\n",
      "  unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n",
      "  running build_src\n",
      "  build_src\n",
      "  building py_modules sources\n",
      "  creating build\n",
      "  creating build\\src.win-amd64-3.8\n",
      "  creating build\\src.win-amd64-3.8\\numpy\n",
      "  creating build\\src.win-amd64-3.8\\numpy\\distutils\n",
      "  building library \"npymath\" sources\n",
      "  get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "  customize GnuFCompiler\n",
      "  customize IntelVisualFCompiler\n",
      "  customize AbsoftFCompiler\n",
      "  customize CompaqVisualFCompiler\n",
      "  customize IntelItaniumVisualFCompiler\n",
      "  customize Gnu95FCompiler\n",
      "  customize Gnu95FCompiler\n",
      "  customize Gnu95FCompiler using config\n",
      "  Traceback (most recent call last):\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\mingw32ccompiler.py\", line 565, in msvc_manifest_xml\n",
      "      fullver = _MSVCRVER_TO_FULLVER[str(maj * 10 + min)]\n",
      "  KeyError: '140'\n",
      "  \n",
      "  During handling of the above exception, another exception occurred:\n",
      "  \n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\setup.py\", line 419, in <module>\n",
      "      setup_package()\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\setup.py\", line 411, in setup_package\n",
      "      setup(**metadata)\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\core.py\", line 171, in setup\n",
      "      return old_setup(**new_attr)\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 153, in setup\n",
      "      return distutils.core.setup(**attrs)\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\core.py\", line 148, in setup\n",
      "      dist.run_commands()\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py\", line 966, in run_commands\n",
      "      self.run_command(cmd)\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\site-packages\\wheel\\bdist_wheel.py\", line 299, in run\n",
      "      self.run_command('build')\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\cmd.py\", line 313, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build.py\", line 47, in run\n",
      "      old_build.run(self)\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\command\\build.py\", line 135, in run\n",
      "      self.run_command(cmd_name)\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\cmd.py\", line 313, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build_src.py\", line 148, in run\n",
      "      self.build_sources()\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build_src.py\", line 159, in build_sources\n",
      "      self.build_library_sources(*libname_info)\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build_src.py\", line 292, in build_library_sources\n",
      "      sources = self.generate_sources(sources, (lib_name, build_info))\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build_src.py\", line 375, in generate_sources\n",
      "      source = func(extension, build_dir)\n",
      "    File \"numpy\\core\\setup.py\", line 665, in get_mathlib_info\n",
      "      st = config_cmd.try_link('int main(void) { return 0;}')\n",
      "    File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\command\\config.py\", line 243, in try_link\n",
      "      self._link(body, headers, include_dirs,\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\config.py\", line 161, in _link\n",
      "      generate_manifest(self)\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\mingw32ccompiler.py\", line 652, in generate_manifest\n",
      "      manxml = msvc_manifest_xml(ma, mi)\n",
      "    File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\mingw32ccompiler.py\", line 567, in msvc_manifest_xml\n",
      "      raise ValueError(\"Version %d,%d of MSVCRT not supported yet\" %\n",
      "  ValueError: Version 14,0 of MSVCRT not supported yet\n",
      "  ----------------------------------------\n",
      "  ERROR: Failed building wheel for numpy\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'c:\\users\\caell\\new_anaconda3\\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\caell\\\\AppData\\\\Local\\\\Temp\\\\pip-install-rg58yztv\\\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\caell\\\\AppData\\\\Local\\\\Temp\\\\pip-install-rg58yztv\\\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' clean --all\n",
      "       cwd: C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\n",
      "  Complete output (10 lines):\n",
      "  Running from numpy source directory.\n",
      "  \n",
      "  `setup.py clean` is not supported, use one of the following instead:\n",
      "  \n",
      "    - `git clean -xdf` (cleans all files)\n",
      "    - `git clean -Xdf` (cleans all versioned files, doesn't touch\n",
      "                        files that aren't checked into the git repo)\n",
      "  \n",
      "  Add `--force` to your command to use it anyway if you must (unsupported).\n",
      "  \n",
      "  ----------------------------------------\n",
      "  ERROR: Failed cleaning build dir for numpy\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'c:\\users\\caell\\new_anaconda3\\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\caell\\\\AppData\\\\Local\\\\Temp\\\\pip-install-rg58yztv\\\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\caell\\\\AppData\\\\Local\\\\Temp\\\\pip-install-rg58yztv\\\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\caell\\AppData\\Local\\Temp\\pip-record-q6x3mcfj\\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\\users\\caell\\new_anaconda3\\Include\\numpy'\n",
      "         cwd: C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\n",
      "    Complete output (383 lines):\n",
      "    Running from numpy source directory.\n",
      "    \n",
      "    Note: if you need reliable uninstall behavior, then install\n",
      "    with pip instead of using `setup.py install`:\n",
      "    \n",
      "      - `pip install .`       (from a git repo or downloaded source\n",
      "                               release)\n",
      "      - `pip install numpy`   (last NumPy release on PyPi)\n",
      "    \n",
      "    \n",
      "    blas_opt_info:\n",
      "    blas_mkl_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries mkl_rt not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    blis_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries blis not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    openblas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "    customize GnuFCompiler\n",
      "    Could not locate executable g77\n",
      "    Could not locate executable f77\n",
      "    customize IntelVisualFCompiler\n",
      "    Could not locate executable ifort\n",
      "    Could not locate executable ifl\n",
      "    customize AbsoftFCompiler\n",
      "    Could not locate executable f90\n",
      "    customize CompaqVisualFCompiler\n",
      "    Could not locate executable DF\n",
      "    customize IntelItaniumVisualFCompiler\n",
      "    Could not locate executable efl\n",
      "    customize Gnu95FCompiler\n",
      "    Found executable C:\\Users\\caell\\new_Anaconda3\\Library\\mingw-w64\\bin\\gfortran.exe\n",
      "    Using built-in specs.\n",
      "    COLLECT_GCC=C:\\Users\\caell\\new_Anaconda3\\Library\\mingw-w64\\bin\\gfortran.exe\n",
      "    COLLECT_LTO_WRAPPER=C:/Users/caell/new_Anaconda3/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/lto-wrapper.exe\n",
      "    Target: x86_64-w64-mingw32\n",
      "    Configured with: ../gcc-5.3.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --with-gxx-include-dir=/mingw64/include/c++/5.3.0 --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,objc,obj-c++,fortran,ada --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --enable-version-specific-runtime-libs --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev5, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld\n",
      "    Thread model: posix\n",
      "    gcc version 5.3.0 (Rev5, Built by MSYS2 project)\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    atlas_3_10_blas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    atlas_3_10_blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    atlas_blas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    atlas_blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    accelerate_info:\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "        Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "        the ATLAS environment variable.\n",
      "      self.calc_info()\n",
      "    blas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries blas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "        Blas (http://www.netlib.org/blas/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [blas]) or by setting\n",
      "        the BLAS environment variable.\n",
      "      self.calc_info()\n",
      "    blas_src_info:\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "        Blas (http://www.netlib.org/blas/) sources not found.\n",
      "        Directories to search for the sources can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [blas_src]) or by setting\n",
      "        the BLAS_SRC environment variable.\n",
      "      self.calc_info()\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    'svnversion' is not recognized as an internal or external command,\n",
      "    operable program or batch file.\n",
      "    non-existing path in 'numpy\\\\distutils': 'site.cfg'\n",
      "    lapack_opt_info:\n",
      "    lapack_mkl_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries mkl_rt not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    openblas_lapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "    customize GnuFCompiler\n",
      "    customize IntelVisualFCompiler\n",
      "    customize AbsoftFCompiler\n",
      "    customize CompaqVisualFCompiler\n",
      "    customize IntelItaniumVisualFCompiler\n",
      "    customize Gnu95FCompiler\n",
      "    Using built-in specs.\n",
      "    COLLECT_GCC=C:\\Users\\caell\\new_Anaconda3\\Library\\mingw-w64\\bin\\gfortran.exe\n",
      "    COLLECT_LTO_WRAPPER=C:/Users/caell/new_Anaconda3/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/lto-wrapper.exe\n",
      "    Target: x86_64-w64-mingw32\n",
      "    Configured with: ../gcc-5.3.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --with-gxx-include-dir=/mingw64/include/c++/5.3.0 --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,objc,obj-c++,fortran,ada --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --enable-version-specific-runtime-libs --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev5, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld\n",
      "    Thread model: posix\n",
      "    gcc version 5.3.0 (Rev5, Built by MSYS2 project)\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    openblas_clapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries openblas,lapack not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "    get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "    customize GnuFCompiler\n",
      "    customize IntelVisualFCompiler\n",
      "    customize AbsoftFCompiler\n",
      "    customize CompaqVisualFCompiler\n",
      "    customize IntelItaniumVisualFCompiler\n",
      "    customize Gnu95FCompiler\n",
      "    Using built-in specs.\n",
      "    COLLECT_GCC=C:\\Users\\caell\\new_Anaconda3\\Library\\mingw-w64\\bin\\gfortran.exe\n",
      "    COLLECT_LTO_WRAPPER=C:/Users/caell/new_Anaconda3/Library/mingw-w64/bin/../lib/gcc/x86_64-w64-mingw32/5.3.0/lto-wrapper.exe\n",
      "    Target: x86_64-w64-mingw32\n",
      "    Configured with: ../gcc-5.3.0/configure --prefix=/mingw64 --with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32 --host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32 --with-native-system-header-dir=/mingw64/x86_64-w64-mingw32/include --libexecdir=/mingw64/lib --with-gxx-include-dir=/mingw64/include/c++/5.3.0 --enable-bootstrap --with-arch=x86-64 --with-tune=generic --enable-languages=c,lto,c++,objc,obj-c++,fortran,ada --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --enable-version-specific-runtime-libs --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64 --with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev5, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld\n",
      "    Thread model: posix\n",
      "    gcc version 5.3.0 (Rev5, Built by MSYS2 project)\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    atlas_3_10_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries tatlas,tatlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "    <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    atlas_3_10_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries satlas,satlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "    <class 'numpy.distutils.system_info.atlas_3_10_info'>\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    atlas_threads_info:\n",
      "    Setting PTATLAS=ATLAS\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries ptf77blas,ptcblas,atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "    <class 'numpy.distutils.system_info.atlas_threads_info'>\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    atlas_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in c:\\users\\caell\\new_anaconda3\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in C:\\\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in c:\\users\\caell\\new_anaconda3\\libs\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack_atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries f77blas,cblas,atlas not found in C:\\Users\\caell\\new_Anaconda3\\Library\\lib\n",
      "    <class 'numpy.distutils.system_info.atlas_info'>\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    lapack_info:\n",
      "    No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "    customize MSVCCompiler\n",
      "      libraries lapack not found in ['c:\\\\users\\\\caell\\\\new_anaconda3\\\\lib', 'C:\\\\', 'c:\\\\users\\\\caell\\\\new_anaconda3\\\\libs', 'C:\\\\Users\\\\caell\\\\new_Anaconda3\\\\Library\\\\lib']\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "        Lapack (http://www.netlib.org/lapack/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [lapack]) or by setting\n",
      "        the LAPACK environment variable.\n",
      "      self.calc_info()\n",
      "    lapack_src_info:\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\system_info.py:639: UserWarning:\n",
      "        Lapack (http://www.netlib.org/lapack/) sources not found.\n",
      "        Directories to search for the sources can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n",
      "        the LAPACK_SRC environment variable.\n",
      "      self.calc_info()\n",
      "      NOT AVAILABLE\n",
      "    \n",
      "    c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py:274: UserWarning: Unknown distribution option: 'define_macros'\n",
      "      warnings.warn(msg)\n",
      "    running install\n",
      "    running build\n",
      "    running config_cc\n",
      "    unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n",
      "    running config_fc\n",
      "    unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n",
      "    running build_src\n",
      "    build_src\n",
      "    building py_modules sources\n",
      "    building library \"npymath\" sources\n",
      "    get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "    customize GnuFCompiler\n",
      "    customize IntelVisualFCompiler\n",
      "    customize AbsoftFCompiler\n",
      "    customize CompaqVisualFCompiler\n",
      "    customize IntelItaniumVisualFCompiler\n",
      "    customize Gnu95FCompiler\n",
      "    customize Gnu95FCompiler\n",
      "    customize Gnu95FCompiler using config\n",
      "    Traceback (most recent call last):\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\mingw32ccompiler.py\", line 565, in msvc_manifest_xml\n",
      "        fullver = _MSVCRVER_TO_FULLVER[str(maj * 10 + min)]\n",
      "    KeyError: '140'\n",
      "    \n",
      "    During handling of the above exception, another exception occurred:\n",
      "    \n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\setup.py\", line 419, in <module>\n",
      "        setup_package()\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\setup.py\", line 411, in setup_package\n",
      "        setup(**metadata)\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\core.py\", line 171, in setup\n",
      "        return old_setup(**new_attr)\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\site-packages\\setuptools\\__init__.py\", line 153, in setup\n",
      "        return distutils.core.setup(**attrs)\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\core.py\", line 148, in setup\n",
      "        dist.run_commands()\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py\", line 966, in run_commands\n",
      "        self.run_command(cmd)\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\install.py\", line 62, in run\n",
      "        r = self.setuptools_run()\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\install.py\", line 36, in setuptools_run\n",
      "        return distutils_install.run(self)\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\command\\install.py\", line 545, in run\n",
      "        self.run_command('build')\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\cmd.py\", line 313, in run_command\n",
      "        self.distribution.run_command(command)\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build.py\", line 47, in run\n",
      "        old_build.run(self)\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\command\\build.py\", line 135, in run\n",
      "        self.run_command(cmd_name)\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\cmd.py\", line 313, in run_command\n",
      "        self.distribution.run_command(command)\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build_src.py\", line 148, in run\n",
      "        self.build_sources()\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build_src.py\", line 159, in build_sources\n",
      "        self.build_library_sources(*libname_info)\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build_src.py\", line 292, in build_library_sources\n",
      "        sources = self.generate_sources(sources, (lib_name, build_info))\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\build_src.py\", line 375, in generate_sources\n",
      "        source = func(extension, build_dir)\n",
      "      File \"numpy\\core\\setup.py\", line 665, in get_mathlib_info\n",
      "        st = config_cmd.try_link('int main(void) { return 0;}')\n",
      "      File \"c:\\users\\caell\\new_anaconda3\\lib\\distutils\\command\\config.py\", line 243, in try_link\n",
      "        self._link(body, headers, include_dirs,\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\command\\config.py\", line 161, in _link\n",
      "        generate_manifest(self)\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\mingw32ccompiler.py\", line 652, in generate_manifest\n",
      "        manxml = msvc_manifest_xml(ma, mi)\n",
      "      File \"C:\\Users\\caell\\AppData\\Local\\Temp\\pip-install-rg58yztv\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\numpy\\distutils\\mingw32ccompiler.py\", line 567, in msvc_manifest_xml\n",
      "        raise ValueError(\"Version %d,%d of MSVCRT not supported yet\" %\n",
      "    ValueError: Version 14,0 of MSVCRT not supported yet\n",
      "    ----------------------------------------\n",
      "ERROR: Command errored out with exit status 1: 'c:\\users\\caell\\new_anaconda3\\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\caell\\\\AppData\\\\Local\\\\Temp\\\\pip-install-rg58yztv\\\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\caell\\\\AppData\\\\Local\\\\Temp\\\\pip-install-rg58yztv\\\\numpy_8e5bbd8a9932478ea962068d49fc89f1\\\\setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\caell\\AppData\\Local\\Temp\\pip-record-q6x3mcfj\\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\\users\\caell\\new_anaconda3\\Include\\numpy' Check the logs for full command output.\n",
      "WARNING: You are using pip version 21.0; however, version 21.0.1 is available.\n",
      "You should consider upgrading via the 'c:\\users\\caell\\new_anaconda3\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Attempting uninstall: numpy\n",
      "    Found existing installation: numpy 1.18.5\n",
      "    Uninstalling numpy-1.18.5:\n",
      "      Successfully uninstalled numpy-1.18.5\n",
      "    Running setup.py install for numpy: started\n",
      "    Running setup.py install for numpy: finished with status 'error'\n",
      "  Rolling back uninstall of numpy\n",
      "  Moving to c:\\users\\caell\\new_anaconda3\\lib\\site-packages\\numpy\n",
      "   from C:\\Users\\caell\\new_Anaconda3\\Lib\\site-packages\\~~mpy\n",
      "  Moving to c:\\users\\caell\\new_anaconda3\\lib\\site-packages\\numpy-1.18.5-py3.8.egg-info\n",
      "   from C:\\Users\\caell\\new_Anaconda3\\Lib\\site-packages\\~-mpy-1.18.5-py3.8.egg-info\n",
      "  Moving to c:\\users\\caell\\new_anaconda3\\scripts\\f2py-script.py\n",
      "   from C:\\Users\\caell\\AppData\\Local\\Temp\\pip-uninstall-643r3kfl\\f2py-script.py\n",
      "  Moving to c:\\users\\caell\\new_anaconda3\\scripts\\f2py.exe\n",
      "   from C:\\Users\\caell\\AppData\\Local\\Temp\\pip-uninstall-643r3kfl\\f2py.exe\n",
      "Requirement already satisfied: dask in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (2021.1.1)\n",
      "Collecting dask\n",
      "  Using cached dask-2021.1.1-py3-none-any.whl (891 kB)\n",
      "  Using cached dask-2021.1.0-py3-none-any.whl (889 kB)\n",
      "Requirement already satisfied: pyyaml in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from dask) (5.3.1)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 21.0; however, version 21.0.1 is available.\n",
      "You should consider upgrading via the 'c:\\users\\caell\\new_anaconda3\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: autogluon in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (0.0.15)\n",
      "Requirement already satisfied: requests in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (2.18.4)\n",
      "Requirement already satisfied: graphviz<0.9.0,>=0.8.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.8.4)\n",
      "Requirement already satisfied: distributed>=2.6.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (2.20.0)\n",
      "Requirement already satisfied: scikit-optimize in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.8.1)\n",
      "Requirement already satisfied: paramiko>=2.4 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (2.7.1)\n",
      "Requirement already satisfied: lightgbm<4.0,>=3.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (3.1.1)\n",
      "Requirement already satisfied: pyarrow<=1.0.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (1.0.0)\n",
      "Requirement already satisfied: ConfigSpace<=0.4.10 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.4.10)\n",
      "Requirement already satisfied: gluoncv<0.9.0,>=0.5.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.8.0)\n",
      "Requirement already satisfied: scipy>=1.3.3 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (1.6.0)\n",
      "Requirement already satisfied: pytest in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (5.4.3)\n",
      "Requirement already satisfied: scikit-learn<0.24,>=0.22.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.23.2)\n",
      "Requirement already satisfied: fastparquet==0.4.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.4.1)\n",
      "Requirement already satisfied: tqdm>=4.38.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (4.47.0)\n",
      "Requirement already satisfied: pandas<2.0,>=1.0.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (1.2.0)\n",
      "Requirement already satisfied: autogluon-contrib-nlp in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.0.1b20210112)\n",
      "Requirement already satisfied: catboost<0.25,>=0.23.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.24.4)\n",
      "Requirement already satisfied: numpy>=1.16.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (1.18.5)\n",
      "Requirement already satisfied: psutil<=5.7.0,>=5.0.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (5.7.0)\n",
      "Requirement already satisfied: cython in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.29.21)\n",
      "Requirement already satisfied: cryptography>=2.8 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (2.9.2)\n",
      "Requirement already satisfied: dask>=2.6.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (2021.1.1)\n",
      "Requirement already satisfied: networkx<3.0,>=2.3 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (2.4)\n",
      "Requirement already satisfied: matplotlib in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (3.2.2)\n",
      "Requirement already satisfied: tornado>=5.0.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (6.1)\n",
      "Requirement already satisfied: openml in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (0.11.0)\n",
      "Requirement already satisfied: boto3 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (1.16.52)\n",
      "Requirement already satisfied: Pillow<=6.2.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon) (6.2.1)\n",
      "Requirement already satisfied: thrift>=0.11.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from fastparquet==0.4.1->autogluon) (0.13.0)\n",
      "Requirement already satisfied: numba>=0.28 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from fastparquet==0.4.1->autogluon) (0.50.1)\n",
      "Requirement already satisfied: packaging in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from fastparquet==0.4.1->autogluon) (20.4)\n",
      "Requirement already satisfied: six in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from fastparquet==0.4.1->autogluon) (1.15.0)\n",
      "Requirement already satisfied: plotly in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from catboost<0.25,>=0.23.0->autogluon) (4.12.0)\n",
      "Requirement already satisfied: typing in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ConfigSpace<=0.4.10->autogluon) (3.7.4.3)\n",
      "Requirement already satisfied: pyparsing in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ConfigSpace<=0.4.10->autogluon) (2.4.7)\n",
      "Requirement already satisfied: cffi!=1.11.3,>=1.8 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from cryptography>=2.8->autogluon) (1.14.0)\n",
      "Requirement already satisfied: pycparser in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from cffi!=1.11.3,>=1.8->cryptography>=2.8->autogluon) (2.20)\n",
      "Requirement already satisfied: pyyaml in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from dask>=2.6.0->autogluon) (5.3.1)\n",
      "Requirement already satisfied: sortedcontainers!=2.0.0,!=2.0.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from distributed>=2.6.0->autogluon) (2.2.2)\n",
      "Requirement already satisfied: tblib>=1.6.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from distributed>=2.6.0->autogluon) (1.6.0)\n",
      "Requirement already satisfied: zict>=0.1.3 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from distributed>=2.6.0->autogluon) (2.0.0)\n",
      "Requirement already satisfied: msgpack>=0.6.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from distributed>=2.6.0->autogluon) (1.0.0)\n",
      "Requirement already satisfied: setuptools in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from distributed>=2.6.0->autogluon) (52.0.0)\n",
      "Requirement already satisfied: toolz>=0.8.2 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from distributed>=2.6.0->autogluon) (0.10.0)\n",
      "Requirement already satisfied: click>=6.6 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from distributed>=2.6.0->autogluon) (7.1.2)\n",
      "Requirement already satisfied: cloudpickle>=1.3.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from distributed>=2.6.0->autogluon) (1.5.0)\n",
      "Requirement already satisfied: portalocker in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from gluoncv<0.9.0,>=0.5.0->autogluon) (2.1.0)\n",
      "Requirement already satisfied: wheel in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from lightgbm<4.0,>=3.0->autogluon) (0.36.2)\n",
      "Requirement already satisfied: decorator>=4.3.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from networkx<3.0,>=2.3->autogluon) (4.4.2)\n",
      "Requirement already satisfied: llvmlite<0.34,>=0.33.0.dev0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from numba>=0.28->fastparquet==0.4.1->autogluon) (0.33.0+1.g022ab0f)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pandas<2.0,>=1.0.0->autogluon) (2.8.1)\n",
      "Requirement already satisfied: pytz>=2017.3 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pandas<2.0,>=1.0.0->autogluon) (2019.3)\n",
      "Requirement already satisfied: bcrypt>=3.1.3 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from paramiko>=2.4->autogluon) (3.1.7)\n",
      "Requirement already satisfied: pynacl>=1.0.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from paramiko>=2.4->autogluon) (1.4.0)\n",
      "Requirement already satisfied: joblib>=0.11 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from scikit-learn<0.24,>=0.22.0->autogluon) (0.16.0)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from scikit-learn<0.24,>=0.22.0->autogluon) (2.1.0)\n",
      "Requirement already satisfied: heapdict in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from zict>=0.1.3->distributed>=2.6.0->autogluon) (1.0.1)\n",
      "Requirement already satisfied: flake8 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon-contrib-nlp->autogluon) (3.8.3)\n",
      "Requirement already satisfied: regex in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon-contrib-nlp->autogluon) (2020.6.8)\n",
      "Requirement already satisfied: tokenizers==0.9.4 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon-contrib-nlp->autogluon) (0.9.4)\n",
      "Requirement already satisfied: yacs>=0.1.6 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon-contrib-nlp->autogluon) (0.1.8)\n",
      "Requirement already satisfied: sentencepiece in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon-contrib-nlp->autogluon) (0.1.95)\n",
      "Requirement already satisfied: sacremoses>=0.0.38 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon-contrib-nlp->autogluon) (0.0.43)\n",
      "Requirement already satisfied: protobuf in c:\\users\\caell\\appdata\\roaming\\python\\python38\\site-packages (from autogluon-contrib-nlp->autogluon) (3.13.0)\n",
      "Requirement already satisfied: contextvars in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon-contrib-nlp->autogluon) (2.4)\n",
      "Requirement already satisfied: sacrebleu in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from autogluon-contrib-nlp->autogluon) (1.5.0)\n",
      "Requirement already satisfied: botocore<1.20.0,>=1.19.52 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from boto3->autogluon) (1.19.52)\n",
      "Requirement already satisfied: s3transfer<0.4.0,>=0.3.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from boto3->autogluon) (0.3.4)\n",
      "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from boto3->autogluon) (0.10.0)\n",
      "Collecting urllib3<1.27,>=1.25.4\n",
      "  Using cached urllib3-1.26.3-py2.py3-none-any.whl (137 kB)\n",
      "Requirement already satisfied: immutables>=0.9 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from contextvars->autogluon-contrib-nlp->autogluon) (0.14)\n",
      "Requirement already satisfied: pycodestyle<2.7.0,>=2.6.0a1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from flake8->autogluon-contrib-nlp->autogluon) (2.6.0)\n",
      "Requirement already satisfied: mccabe<0.7.0,>=0.6.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from flake8->autogluon-contrib-nlp->autogluon) (0.6.1)\n",
      "Requirement already satisfied: pyflakes<2.3.0,>=2.2.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from flake8->autogluon-contrib-nlp->autogluon) (2.2.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from matplotlib->autogluon) (1.2.0)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from matplotlib->autogluon) (0.10.0)\n",
      "Requirement already satisfied: xmltodict in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from openml->autogluon) (0.12.0)\n",
      "Requirement already satisfied: liac-arff>=2.4.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from openml->autogluon) (2.5.0)\n",
      "Requirement already satisfied: retrying>=1.3.3 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from plotly->catboost<0.25,>=0.23.0->autogluon) (1.3.3)\n",
      "Requirement already satisfied: pywin32!=226 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from portalocker->gluoncv<0.9.0,>=0.5.0->autogluon) (227)\n",
      "Requirement already satisfied: py>=1.5.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pytest->autogluon) (1.9.0)\n",
      "Requirement already satisfied: attrs>=17.4.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pytest->autogluon) (19.3.0)\n",
      "Requirement already satisfied: more-itertools>=4.0.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pytest->autogluon) (8.4.0)\n",
      "Requirement already satisfied: pluggy<1.0,>=0.12 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pytest->autogluon) (0.13.1)\n",
      "Requirement already satisfied: wcwidth in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pytest->autogluon) (0.2.5)\n",
      "Requirement already satisfied: atomicwrites>=1.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pytest->autogluon) (1.4.0)\n",
      "Requirement already satisfied: colorama in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from pytest->autogluon) (0.4.3)\n",
      "Requirement already satisfied: idna<2.7,>=2.5 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from requests->autogluon) (2.6)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from requests->autogluon) (2020.12.5)\n",
      "Collecting requests\n",
      "  Using cached requests-2.25.1-py2.py3-none-any.whl (61 kB)\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from requests->autogluon) (3.0.4)\n",
      "Requirement already satisfied: pyaml>=16.9 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from scikit-optimize->autogluon) (20.4.0)\n",
      "Installing collected packages: urllib3, requests\n",
      "  Attempting uninstall: urllib3\n",
      "    Found existing installation: urllib3 1.22\n",
      "    Uninstalling urllib3-1.22:\n",
      "      Successfully uninstalled urllib3-1.22\n",
      "  Attempting uninstall: requests\n",
      "    Found existing installation: requests 2.18.4\n",
      "    Uninstalling requests-2.18.4:\n",
      "      Successfully uninstalled requests-2.18.4\n",
      "Successfully installed requests-2.25.1 urllib3-1.26.3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "pmdarima 1.7.1 requires Cython<0.29.18,>=0.29, but you have cython 0.29.21 which is incompatible.\n",
      "pmdarima 1.7.1 requires setuptools<50.0.0, but you have setuptools 52.0.0 which is incompatible.\n",
      "gensim 3.8.3 requires Cython==0.29.14, but you have cython 0.29.21 which is incompatible.\n",
      "WARNING: You are using pip version 21.0; however, version 21.0.1 is available.\n",
      "You should consider upgrading via the 'c:\\users\\caell\\new_anaconda3\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: ipykernel in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (5.4.3)\n",
      "Collecting ipykernel\n",
      "  Using cached ipykernel-5.4.3-py3-none-any.whl (120 kB)\n",
      "  Using cached ipykernel-5.4.2-py3-none-any.whl (119 kB)\n",
      "Requirement already satisfied: tornado>=4.2 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipykernel) (6.1)\n",
      "Requirement already satisfied: jupyter-client in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipykernel) (6.1.6)\n",
      "Requirement already satisfied: traitlets>=4.1.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipykernel) (4.3.3)\n",
      "Requirement already satisfied: ipython>=5.0.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipykernel) (7.16.1)\n",
      "Requirement already satisfied: jedi>=0.10 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipython>=5.0.0->ipykernel) (0.17.1)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipython>=5.0.0->ipykernel) (3.0.5)\n",
      "Requirement already satisfied: decorator in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipython>=5.0.0->ipykernel) (4.4.2)\n",
      "Requirement already satisfied: pygments in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipython>=5.0.0->ipykernel) (2.6.1)\n",
      "Requirement already satisfied: backcall in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipython>=5.0.0->ipykernel) (0.2.0)\n",
      "Requirement already satisfied: pickleshare in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipython>=5.0.0->ipykernel) (0.7.5)\n",
      "Requirement already satisfied: colorama in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipython>=5.0.0->ipykernel) (0.4.3)\n",
      "Requirement already satisfied: setuptools>=18.5 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from ipython>=5.0.0->ipykernel) (52.0.0)\n",
      "Requirement already satisfied: parso<0.8.0,>=0.7.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from jedi>=0.10->ipython>=5.0.0->ipykernel) (0.7.0)\n",
      "Requirement already satisfied: wcwidth in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=5.0.0->ipykernel) (0.2.5)\n",
      "Requirement already satisfied: ipython-genutils in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from traitlets>=4.1.0->ipykernel) (0.2.0)\n",
      "Requirement already satisfied: six in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from traitlets>=4.1.0->ipykernel) (1.15.0)\n",
      "Requirement already satisfied: python-dateutil>=2.1 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from jupyter-client->ipykernel) (2.8.1)\n",
      "Requirement already satisfied: jupyter-core>=4.6.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from jupyter-client->ipykernel) (4.6.3)\n",
      "Requirement already satisfied: pyzmq>=13 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from jupyter-client->ipykernel) (19.0.1)\n",
      "Requirement already satisfied: pywin32>=1.0 in c:\\users\\caell\\new_anaconda3\\lib\\site-packages (from jupyter-core>=4.6.0->jupyter-client->ipykernel) (227)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: You are using pip version 21.0; however, version 21.0.1 is available.\n",
      "You should consider upgrading via the 'c:\\users\\caell\\new_anaconda3\\python.exe -m pip install --upgrade pip' command.\n"
     ]
    },
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'mxnet'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-11-1f65fd6b92a6>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m     16\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     17\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 18\u001b[1;33m \u001b[1;32mimport\u001b[0m \u001b[0mmxnet\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mmx\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     19\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mmxnet\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mndarray\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mnd\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     20\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mmxnet\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mnd\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mautograd\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mgluon\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'mxnet'"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Uninstall mkl for faster neural-network training time\n",
    "# Upgrade pip to ensure the latest package versions are available\n",
    "!pip install mxnet --upgrade\n",
    "!pip install dask --upgrade\n",
    "# !pip install -U \"mxnet<2.0.0\"\n",
    "!pip install autogluon\n",
    "!pip install -U ipykernel\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "np.random.seed(111)\n",
    "from __future__ import print_function\n",
    "\n",
    "\n",
    "import mxnet as mx\n",
    "import mxnet.ndarray as nd\n",
    "from mxnet import nd, autograd, gluon\n",
    "from mxnet.gluon.data.vision import transforms\n",
    "from autogluon import TextPrediction as task\n",
    "\n",
    "from src import load_text\n",
    "\n",
    "ctx = mx.gpu() if mx.context.num_gpus() else mx.cpu()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 412
    },
    "id": "mQeSqHKNex-c",
    "outputId": "3f0c9501-e90f-4ee4-b9bb-020de086a87d",
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'load_text' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-10-c6c756dd94b3>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mdf\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mload_text\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msentences\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdropna\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'load_text' is not defined"
     ]
    }
   ],
   "source": [
    "df = load_text(sentences=True).dropna()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'task' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-8-c7cebffead99>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m predictor = task.fit(train_df[['Text','Grade']], label='Grade',\n\u001b[0m\u001b[0;32m      2\u001b[0m                      \u001b[0mseed\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m111\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m                     )\n",
      "\u001b[1;31mNameError\u001b[0m: name 'task' is not defined"
     ]
    }
   ],
   "source": [
    "\n",
    "predictor = task.fit(train_df[['Text','Grade']], label='Grade',\n",
    "                     seed=111\n",
    "                    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "-Ig0Epi_gsV7"
   },
   "outputs": [],
   "source": [
    "randindx = np.random.permutation(np.arange(len(df)))\n",
    "train_df = df.iloc[randindx[:int(len(randindx)*.8)]]\n",
    "test_df = df.iloc[randindx[len(train_df):]]\n",
    "errors = test_df.copy()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "VoczHnu6mDir"
   },
   "source": [
    "ERROR: distributed 2021.1.1 has requirement dask>=2020.12.0, but you'll have dask 2.12.0 which is incompatible.\n",
    "ERROR: albumentations 0.1.12 has requirement imgaug<0.2.7,>=0.2.5, but you'll have imgaug 0.2.9 which is incompatible.\n",
    "ERROR: botocore 1.19.59 has requirement urllib3<1.27,>=1.25.4; python_version != \"3.4\", but you'll have urllib3 1.24.3 which is incompatible.\n",
    "I"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000,
     "referenced_widgets": [
      "c29059ef29494fa0b3a28ac1c7d85240",
      "27a0ed6ab0ef44e680a1990ae47a833d",
      "b966808e12324120bf64600ce8a2164d",
      "15785b940abb4e21af4d78a84817f7bd",
      "bb95cfd163de4d078a2b68022370f5f3",
      "7b289856bfcd43218f4618cf65963a10",
      "92a2542f14474b48b1455679a89a4935",
      "5e43dbdc5af64d489b795dfab3a3a05c"
     ]
    },
    "id": "y_Rhpt1IjfBq",
    "outputId": "42b72438-9509-45cb-df5c-204533a3600e"
   },
   "outputs": [],
   "source": [
    "\n",
    "predictor = task.fit(train_df[['Text','Grade']], label='Grade',\n",
    "                     seed=111\n",
    "                    )\n",
    "print(predictor.evaluate(test_df[['Text','Grade']], metrics=['mae']))\n",
    "yhat = predictor.predict({'Text':[text for text in test_df.Text]})\n",
    "errors['Text Prediction'] = yhat\n",
    "errors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000,
     "referenced_widgets": [
      "f6a15d277e1f49069a1878a1573e52fd",
      "eaaf130b574140fd80f862ae689f38aa",
      "5c5981276efd4f8ab33a19fa0dcfdf46",
      "0836f898bf07477c9df50fe393eda030",
      "c0e3e032d9604d7f9d1f941b8be42773",
      "2762f9b804ea4b46984653347b20ab59",
      "d7c07ed5e4e2492fad90bac7126ce5e5",
      "fa80465b443c43be81eb459221f06472"
     ]
    },
    "id": "M89BAFnupQrz",
    "outputId": "342a48a2-1369-45ca-b0d7-674f9895c545"
   },
   "outputs": [],
   "source": [
    "predictor2 = task.fit(train_df[['grammarized','Grade']], label='Grade',\n",
    "                     seed=111\n",
    "                    )\n",
    "print(predictor2.evaluate(test_df[['grammarized','Grade']], metrics=['mae']))\n",
    "yhat = predictor2.predict({'grammarized':[text for text in test_df.grammarized]})\n",
    "errors['Grammarized Prediction'] = yhat\n",
    "errors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "xyrTmaPDuY9L"
   },
   "outputs": [],
   "source": [
    "predictor3 = task.fit(train_df[['lemmatized','Grade']], label='Grade',\n",
    "                     seed=111\n",
    "                    )\n",
    "print(predictor3.evaluate(test_df[['lemmatized','Grade']], metrics=['mae']))\n",
    "yhat = predictor3.predict({'lemmatized':[text for text in test_df.Text]})\n",
    "errors['Lemmatized Prediction'] = yhat\n",
    "errors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "95vsh7gvuqhj"
   },
   "outputs": [],
   "source": [
    "errors = errors.iloc[:,[1,2,3,0,4,5]]\n",
    "errors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 161
    },
    "id": "aYAIKbkqO7Km",
    "outputId": "09db3a1e-933a-474f-eaea-181232824f22"
   },
   "outputs": [],
   "source": [
    "predictor.export('TextModel')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "OTppnzQa2-ly"
   },
   "outputs": [],
   "source": [
    "net.export(\"lenet\", epoch=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "b_KCOdEaf5Le"
   },
   "outputs": [],
   "source": [
    "import warnings\n",
    "with warnings.catch_warnings():\n",
    "    warnings.simplefilter(\"ignore\")\n",
    "    deserialized_net = gluon.nn.SymbolBlock.imports(\"lenet-symbol.json\", ['data'], \"lenet-0001.params\", ctx=ctx)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "iS-gryHof7oK"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "accelerator": "GPU",
  "colab": {
   "name": "autogluon_student_writing_models.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "0836f898bf07477c9df50fe393eda030": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "HTMLModel",
     "state": {
      "_dom_classes": [],
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "HTMLModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/controls",
      "_view_module_version": "1.5.0",
      "_view_name": "HTMLView",
      "description": "",
      "description_tooltip": null,
      "layout": "IPY_MODEL_fa80465b443c43be81eb459221f06472",
      "placeholder": "​",
      "style": "IPY_MODEL_d7c07ed5e4e2492fad90bac7126ce5e5",
      "value": " 3/4 [38:32&lt;08:52, 532.03s/it]"
     }
    },
    "15785b940abb4e21af4d78a84817f7bd": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "HTMLModel",
     "state": {
      "_dom_classes": [],
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "HTMLModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/controls",
      "_view_module_version": "1.5.0",
      "_view_name": "HTMLView",
      "description": "",
      "description_tooltip": null,
      "layout": "IPY_MODEL_5e43dbdc5af64d489b795dfab3a3a05c",
      "placeholder": "​",
      "style": "IPY_MODEL_92a2542f14474b48b1455679a89a4935",
      "value": " 4/4 [1:08:35&lt;00:00, 1028.82s/it]"
     }
    },
    "2762f9b804ea4b46984653347b20ab59": {
     "model_module": "@jupyter-widgets/base",
     "model_name": "LayoutModel",
     "state": {
      "_model_module": "@jupyter-widgets/base",
      "_model_module_version": "1.2.0",
      "_model_name": "LayoutModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "LayoutView",
      "align_content": null,
      "align_items": null,
      "align_self": null,
      "border": null,
      "bottom": null,
      "display": null,
      "flex": null,
      "flex_flow": null,
      "grid_area": null,
      "grid_auto_columns": null,
      "grid_auto_flow": null,
      "grid_auto_rows": null,
      "grid_column": null,
      "grid_gap": null,
      "grid_row": null,
      "grid_template_areas": null,
      "grid_template_columns": null,
      "grid_template_rows": null,
      "height": null,
      "justify_content": null,
      "justify_items": null,
      "left": null,
      "margin": null,
      "max_height": null,
      "max_width": null,
      "min_height": null,
      "min_width": null,
      "object_fit": null,
      "object_position": null,
      "order": null,
      "overflow": null,
      "overflow_x": null,
      "overflow_y": null,
      "padding": null,
      "right": null,
      "top": null,
      "visibility": null,
      "width": null
     }
    },
    "27a0ed6ab0ef44e680a1990ae47a833d": {
     "model_module": "@jupyter-widgets/base",
     "model_name": "LayoutModel",
     "state": {
      "_model_module": "@jupyter-widgets/base",
      "_model_module_version": "1.2.0",
      "_model_name": "LayoutModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "LayoutView",
      "align_content": null,
      "align_items": null,
      "align_self": null,
      "border": null,
      "bottom": null,
      "display": null,
      "flex": null,
      "flex_flow": null,
      "grid_area": null,
      "grid_auto_columns": null,
      "grid_auto_flow": null,
      "grid_auto_rows": null,
      "grid_column": null,
      "grid_gap": null,
      "grid_row": null,
      "grid_template_areas": null,
      "grid_template_columns": null,
      "grid_template_rows": null,
      "height": null,
      "justify_content": null,
      "justify_items": null,
      "left": null,
      "margin": null,
      "max_height": null,
      "max_width": null,
      "min_height": null,
      "min_width": null,
      "object_fit": null,
      "object_position": null,
      "order": null,
      "overflow": null,
      "overflow_x": null,
      "overflow_y": null,
      "padding": null,
      "right": null,
      "top": null,
      "visibility": null,
      "width": null
     }
    },
    "5c5981276efd4f8ab33a19fa0dcfdf46": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "FloatProgressModel",
     "state": {
      "_dom_classes": [],
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "FloatProgressModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/controls",
      "_view_module_version": "1.5.0",
      "_view_name": "ProgressView",
      "bar_style": "danger",
      "description": " 75%",
      "description_tooltip": null,
      "layout": "IPY_MODEL_2762f9b804ea4b46984653347b20ab59",
      "max": 4,
      "min": 0,
      "orientation": "horizontal",
      "style": "IPY_MODEL_c0e3e032d9604d7f9d1f941b8be42773",
      "value": 3
     }
    },
    "5e43dbdc5af64d489b795dfab3a3a05c": {
     "model_module": "@jupyter-widgets/base",
     "model_name": "LayoutModel",
     "state": {
      "_model_module": "@jupyter-widgets/base",
      "_model_module_version": "1.2.0",
      "_model_name": "LayoutModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "LayoutView",
      "align_content": null,
      "align_items": null,
      "align_self": null,
      "border": null,
      "bottom": null,
      "display": null,
      "flex": null,
      "flex_flow": null,
      "grid_area": null,
      "grid_auto_columns": null,
      "grid_auto_flow": null,
      "grid_auto_rows": null,
      "grid_column": null,
      "grid_gap": null,
      "grid_row": null,
      "grid_template_areas": null,
      "grid_template_columns": null,
      "grid_template_rows": null,
      "height": null,
      "justify_content": null,
      "justify_items": null,
      "left": null,
      "margin": null,
      "max_height": null,
      "max_width": null,
      "min_height": null,
      "min_width": null,
      "object_fit": null,
      "object_position": null,
      "order": null,
      "overflow": null,
      "overflow_x": null,
      "overflow_y": null,
      "padding": null,
      "right": null,
      "top": null,
      "visibility": null,
      "width": null
     }
    },
    "7b289856bfcd43218f4618cf65963a10": {
     "model_module": "@jupyter-widgets/base",
     "model_name": "LayoutModel",
     "state": {
      "_model_module": "@jupyter-widgets/base",
      "_model_module_version": "1.2.0",
      "_model_name": "LayoutModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "LayoutView",
      "align_content": null,
      "align_items": null,
      "align_self": null,
      "border": null,
      "bottom": null,
      "display": null,
      "flex": null,
      "flex_flow": null,
      "grid_area": null,
      "grid_auto_columns": null,
      "grid_auto_flow": null,
      "grid_auto_rows": null,
      "grid_column": null,
      "grid_gap": null,
      "grid_row": null,
      "grid_template_areas": null,
      "grid_template_columns": null,
      "grid_template_rows": null,
      "height": null,
      "justify_content": null,
      "justify_items": null,
      "left": null,
      "margin": null,
      "max_height": null,
      "max_width": null,
      "min_height": null,
      "min_width": null,
      "object_fit": null,
      "object_position": null,
      "order": null,
      "overflow": null,
      "overflow_x": null,
      "overflow_y": null,
      "padding": null,
      "right": null,
      "top": null,
      "visibility": null,
      "width": null
     }
    },
    "92a2542f14474b48b1455679a89a4935": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "DescriptionStyleModel",
     "state": {
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "DescriptionStyleModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "StyleView",
      "description_width": ""
     }
    },
    "b966808e12324120bf64600ce8a2164d": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "FloatProgressModel",
     "state": {
      "_dom_classes": [],
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "FloatProgressModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/controls",
      "_view_module_version": "1.5.0",
      "_view_name": "ProgressView",
      "bar_style": "success",
      "description": "100%",
      "description_tooltip": null,
      "layout": "IPY_MODEL_7b289856bfcd43218f4618cf65963a10",
      "max": 4,
      "min": 0,
      "orientation": "horizontal",
      "style": "IPY_MODEL_bb95cfd163de4d078a2b68022370f5f3",
      "value": 4
     }
    },
    "bb95cfd163de4d078a2b68022370f5f3": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "ProgressStyleModel",
     "state": {
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "ProgressStyleModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "StyleView",
      "bar_color": null,
      "description_width": "initial"
     }
    },
    "c0e3e032d9604d7f9d1f941b8be42773": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "ProgressStyleModel",
     "state": {
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "ProgressStyleModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "StyleView",
      "bar_color": null,
      "description_width": "initial"
     }
    },
    "c29059ef29494fa0b3a28ac1c7d85240": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "HBoxModel",
     "state": {
      "_dom_classes": [],
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "HBoxModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/controls",
      "_view_module_version": "1.5.0",
      "_view_name": "HBoxView",
      "box_style": "",
      "children": [
       "IPY_MODEL_b966808e12324120bf64600ce8a2164d",
       "IPY_MODEL_15785b940abb4e21af4d78a84817f7bd"
      ],
      "layout": "IPY_MODEL_27a0ed6ab0ef44e680a1990ae47a833d"
     }
    },
    "d7c07ed5e4e2492fad90bac7126ce5e5": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "DescriptionStyleModel",
     "state": {
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "DescriptionStyleModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "StyleView",
      "description_width": ""
     }
    },
    "eaaf130b574140fd80f862ae689f38aa": {
     "model_module": "@jupyter-widgets/base",
     "model_name": "LayoutModel",
     "state": {
      "_model_module": "@jupyter-widgets/base",
      "_model_module_version": "1.2.0",
      "_model_name": "LayoutModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "LayoutView",
      "align_content": null,
      "align_items": null,
      "align_self": null,
      "border": null,
      "bottom": null,
      "display": null,
      "flex": null,
      "flex_flow": null,
      "grid_area": null,
      "grid_auto_columns": null,
      "grid_auto_flow": null,
      "grid_auto_rows": null,
      "grid_column": null,
      "grid_gap": null,
      "grid_row": null,
      "grid_template_areas": null,
      "grid_template_columns": null,
      "grid_template_rows": null,
      "height": null,
      "justify_content": null,
      "justify_items": null,
      "left": null,
      "margin": null,
      "max_height": null,
      "max_width": null,
      "min_height": null,
      "min_width": null,
      "object_fit": null,
      "object_position": null,
      "order": null,
      "overflow": null,
      "overflow_x": null,
      "overflow_y": null,
      "padding": null,
      "right": null,
      "top": null,
      "visibility": null,
      "width": null
     }
    },
    "f6a15d277e1f49069a1878a1573e52fd": {
     "model_module": "@jupyter-widgets/controls",
     "model_name": "HBoxModel",
     "state": {
      "_dom_classes": [],
      "_model_module": "@jupyter-widgets/controls",
      "_model_module_version": "1.5.0",
      "_model_name": "HBoxModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/controls",
      "_view_module_version": "1.5.0",
      "_view_name": "HBoxView",
      "box_style": "",
      "children": [
       "IPY_MODEL_5c5981276efd4f8ab33a19fa0dcfdf46",
       "IPY_MODEL_0836f898bf07477c9df50fe393eda030"
      ],
      "layout": "IPY_MODEL_eaaf130b574140fd80f862ae689f38aa"
     }
    },
    "fa80465b443c43be81eb459221f06472": {
     "model_module": "@jupyter-widgets/base",
     "model_name": "LayoutModel",
     "state": {
      "_model_module": "@jupyter-widgets/base",
      "_model_module_version": "1.2.0",
      "_model_name": "LayoutModel",
      "_view_count": null,
      "_view_module": "@jupyter-widgets/base",
      "_view_module_version": "1.2.0",
      "_view_name": "LayoutView",
      "align_content": null,
      "align_items": null,
      "align_self": null,
      "border": null,
      "bottom": null,
      "display": null,
      "flex": null,
      "flex_flow": null,
      "grid_area": null,
      "grid_auto_columns": null,
      "grid_auto_flow": null,
      "grid_auto_rows": null,
      "grid_column": null,
      "grid_gap": null,
      "grid_row": null,
      "grid_template_areas": null,
      "grid_template_columns": null,
      "grid_template_rows": null,
      "height": null,
      "justify_content": null,
      "justify_items": null,
      "left": null,
      "margin": null,
      "max_height": null,
      "max_width": null,
      "min_height": null,
      "min_width": null,
      "object_fit": null,
      "object_position": null,
      "order": null,
      "overflow": null,
      "overflow_x": null,
      "overflow_y": null,
      "padding": null,
      "right": null,
      "top": null,
      "visibility": null,
      "width": null
     }
    }
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
